src/containers/Dashboard.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Chart } from 'primereact/chart';
+import '@binaris/shift-babel-macro/macro';
+
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Container from 'react-bootstrap/Container';
+import Button from 'react-bootstrap/Button';
+
+import {
+  generateRemoteData,
+  setStorageData,
+  getStorageData,
+} from '../../backend/dashboard-server';
+import { Display } from '../constants/enums';
+
+export default class Dashboard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chart_data: {
+        labels: ['A', 'B', 'C'],
+        datasets: [
+          {
+            label: 'My First dataset',
+            data: [100, 200, 300],
+            backgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+            hoverBackgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+          },
+        ],
+      },
+      display: Display.FRONTEND,
+    };
+    this.generateLocalData = this.generateLocalData.bind(this);
+    this.getRemoteData = this.getRemoteData.bind(this);
+    this.fetchFromStorageData = this.fetchFromStorageData.bind(this);
+    this.initStorageData();
```

Comment:
You're not supposed to have side-effects here, use `componentDidMount` instead or better yet `useEffect` hook when rewriting to a functional component.